File: src/providers/cli-dirs.js
#!/usr/bin/env node
/**
 * @module providers/cli-dirs
 * @description Tiny helper called by shell scripts to get CLI adapter directory paths.
 *
 * Usage from bash:
 *   AGENT_DIR=$(node src/providers/cli-dirs.js agentDir)
 *   CONFIG_DIR=$(node src/providers/cli-dirs.js configDir)
 */

const { getCliAdapter } = require('./cli-adapter');

const field = process.argv[2];
const adapter = getCliAdapter();

if (field === 'agentDir') {
  console.log(adapter.agentDir);
} else if (field === 'configDir') {
  console.log(adapter.configDir);
} else if (field === 'rulesFileName') {
  console.log(adapter.rulesFileName);
} else {
  console.error('Usage: node cli-dirs.js agentDir|configDir|rulesFileName');
  process.exit(1);
}

File: scripts/cli_dirs.py
#!/usr/bin/env python3
"""Print CLI adapter directory paths for use by shell scripts.

A small helper that loads the configured CLI adapter and prints one of its
directory-related properties to stdout. Used by shell scripts (e.g.
``start.sh``, ``stop.sh``) that need to know where agent files live without
parsing YAML themselves.

Usage::

    python3 scripts/cli_dirs.py agentDir      # e.g. ".claude/agents"
    python3 scripts/cli_dirs.py configDir     # e.g. ".claude"
    python3 scripts/cli_dirs.py rulesFileName # e.g. "CLAUDE.md"
"""

import sys
import os

sys.path.insert(0, os.path.join(os.path.dirname(__file__), ".."))

from src.providers.cli_adapter import get_cli_adapter

field = sys.argv[1] if len(sys.argv) > 1 else ""
adapter, _ = get_cli_adapter()

if field == "agentDir":
    print(adapter.agent_dir)
elif field == "configDir":
    print(adapter.config_dir)
elif field == "rulesFileName":
    print(adapter.rules_file_name)
else:
    print("Usage: python3 cli_dirs.py agentDir|configDir|rulesFileName", file=sys.stderr)
    sys.exit(1)
